src/replication/status.cpp

```diff
@@ -72,31 +79,42 @@ void from_json(const nlohmann::json &j, ReplicationRoleEntry &p) {
       break;
     }
     case memgraph::replication_coordination_glue::ReplicationRole::REPLICA: {
-      std::string ip_address;
+      std::string address;
       // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
       uint16_t port;
-      j.at(kIpAddress).get_to(ip_address);
+      // In V4 we renamed key `replica_ip_address` to `replica_address`
+      if (version == DurabilityVersion::V4) {
+        j.at(kAddress).get_to(address);
+      } else {
+        j.at(kIpAddress).get_to(address);
+      }
+
       j.at(kPort).get_to(port);
-      auto config = ReplicationServerConfig{.ip_address = std::move(ip_address), .port = port};
+      auto config = ReplicationServerConfig{.address = std::move(address), .port = port};
       auto replica_role = ReplicaRole{.config = std::move(config)};
       if (j.contains(kMainUUID)) {
         replica_role.main_uuid = j.at(kMainUUID);
       }
-
+      // TODO: (andi) Should we bump here to V4? After reading old version, modify and bump to new?
       p = ReplicationRoleEntry{.version = version, .role = std::move(replica_role)};
-
       break;
     }
   }
 }
 
 void to_json(nlohmann::json &j, const ReplicationReplicaEntry &p) {
-  auto common = nlohmann::json{{kReplicaName, p.config.name},
-                               {kIpAddress, p.config.ip_address},
+  auto common = nlohmann::json{{kVersion, p.version},
+                               {kReplicaName, p.config.name},
                                {kPort, p.config.port},
                                {kSyncMode, p.config.mode},
                                {kCheckFrequency, p.config.replica_check_frequency.count()}};
 
+  if (p.version == DurabilityVersion::V4) {
```

Comment:
Again, lets be explicit about the versions.

Reply:
ReplicationReplicaEntry didn't have durability version before. So p.version can never be DurabilityVersion::v1, v2. v3.

That's why I wrote in that way. What do you think?